docs/configuration-reference.md

```diff
@@ -1869,17 +1869,50 @@ metadata_config:
 
 ### loki_config
 
-The `loki_config` block configures how the Agent collects logs and sends them to a Loki push API endpoint. `loki_config` is identical to how Promtail is configured, except deprecated
-fields have been removed and the server_config is not supported.
+The `loki_config` block configures how the Agent collects logs and sends them to
+a Loki push API endpoint. `loki_config` is identical to how Promtail is
+configured, except deprecated fields have been removed and the server_config is
+not supported.
 
 Please refer to the
 [Promtail documentation](https://github.com/grafana/loki/tree/master/docs/sources/clients/promtail#client_config)
 for the supported values for these fields.
 
 ```yaml
+# Directory to store Loki/Promtail positions files in. Positions files are
+# required to read logs, and are used to store the last read offset of log
+# sources. The positions files will be stored in
+# <positions_directory>/<loki_instance_config.name>.yml.
+#
+# Optional only if every config has a positions.filename manually provided.
+[positions_directory: <string>]
+
+# Loki Promtail instances to run for log collection.
+configs:
+  - [<loki_instance_config>]
+```
+
+### loki_instance_config
+
+The `loki_instance_config` block is an individual instance of Promtail with its
+own set of scrape rules and where to forward logs. It is identical to how
+Promtail is configured, except deprecated fields have been removed and the
+`server_config` block is not supported.
+
+Please refer to the
+[Promtail documentation](https://github.com/grafana/loki/tree/master/docs/sources/clients/promtail#client_config)
+for the supported values for these fields.
```

Comment:
Even if promtail is the source of truth for these fields, should the Agent configuration have a small example so that people don't have to context switch? Do we do something similar for Prometheus config?

Reply:
I'd like to eventually generate this section to avoid context switching (#120). Cortex [does something similar for its documentation](https://github.com/cortexproject/cortex/tree/master/tools/doc-generator). IMO until #120 is done, the best we can do is tell people to context switch. I'm concerned we'll end up giving people more questions than answers since there's not really a good example config that'll cover the majority of common use cases.